### PR TITLE
Fix #105

### LINKED ISSUE #105: Fix BigNumber Validation Error During DEX Swap
BigNumber refactoring in conjunction with the new input validation techniques must have slipped by in testing because every DEX swap is failing to work properly again. Likely part of the recent merges. 

Will fix before delivery.

### PR DESCRIPTION
**Pull request by outragedhuman**
Closes #105 

Converts BigNumber values under validation into strings for consistency

### CODE REVIEW

File: src/utils/validation/index.ts [hunk 1]
```diff
@@ -27,41 +28,56 @@ export function isValidAssetID(assetText: string) {
...
+  quantity = bnToString(quantity);
 
   // ensure value is a string
   if (typeof quantity !== "string") {
     console.error("quantity entered is not a string");
     return false;
   }
 
+  // ensure only a single decimal
+  if (quantity.split(".").length > 2) {
```
Comment (vwalsh) [inline src/utils/validation/index.ts:41]: i'm going to run this in the next few days and see if i can break it with some bad inputs. that'll give me confidence that it's working. would be great if i did it with cypress so you had a working test to go with it.

Comment (outragedhuman) [reply src/utils/validation/index.ts:41]: great idea!

**Comment by outragedhuman** [comment]
Note: this was blocked but then needed to be merged into another working branch, that branch has now been merged with master so this PR is no longer needed but it's being left open until the testing is completed just to remind us.

File: src/utils/validation/index.ts (hunk 1 above)
Comment (vwalsh) [reply src/utils/validation/index.ts:41]: upon further review of the bignumber code, tostring doesn't seem to offer up any way to make two decimals exist by using a really large number or really small. see https://mikemcl.github.io/bignumber.js/#toS for details

**Pull request closed without merging**